src/mscorlib/shared/System/Convert.cs

```diff
@@ -258,79 +248,74 @@ public static Object ChangeType(Object value, TypeCode typeCode, IFormatProvider
                 case TypeCode.Object:
                     return value;
                 case TypeCode.DBNull:
-                    throw new InvalidCastException(Environment.GetResourceString("InvalidCast_DBNull"));
+                    throw new InvalidCastException(SR.InvalidCast_DBNull);
                 case TypeCode.Empty:
-                    throw new InvalidCastException(Environment.GetResourceString("InvalidCast_Empty"));
+                    throw new InvalidCastException(SR.InvalidCast_Empty);
                 default:
-                    throw new ArgumentException(Environment.GetResourceString("Arg_UnknownTypeCode"));
+                    throw new ArgumentException(SR.Arg_UnknownTypeCode);
             }
         }
 
         internal static Object DefaultToType(IConvertible value, Type targetType, IFormatProvider provider)
         {
-            Contract.Requires(value != null, "[Convert.DefaultToType]value!=null");
+            Debug.Assert(value != null, "[Convert.DefaultToType]value!=null");
             if (targetType == null)
             {
                 throw new ArgumentNullException(nameof(targetType));
             }
             Contract.EndContractBlock();
 
-            RuntimeType rtTargetType = targetType as RuntimeType;
-
-            if (rtTargetType != null)
```

Comment:
Yes, `ReferenceEquals` would be faster, generate smaller native code and work correctly. 

It is depending on internal implementation detail that there is just one instance of RuntimeType for given type. There is a ton of code out depending on this detail already and thus we are not ever going to be able to change it, so taking one more dependency on it should not be a problem.